specifyweb/frontend/js_src/lib/components/Molecules/GenericSortedDataViewer.tsx

```diff
@@ -0,0 +1,83 @@
+import { SchemaViewerRow, SchemaViewerValue } from '../SchemaViewer/helpers';
+import { RA, RR } from '../../utils/types';
+import { LocalizedString } from 'typesafe-i18n';
+import React from 'react';
+import { Link } from '../Atoms/Link';
+
+export function GenericSortedDataViewer<
+  DATA extends SchemaViewerRow<RR<string, SchemaViewerValue>>
+>({
+  headers,
+  data,
+  getLink,
+  className = '',
+  headerClassName = '',
+  cellClassName,
+}: {
+  readonly headers: RR<string, JSX.Element | LocalizedString>;
+  readonly headerClassName?: string;
+  readonly className?: string;
+  readonly data: RA<DATA>;
+  readonly getLink: ((row: DATA) => string) | undefined;
+  readonly cellClassName?: (
+    row: DATA,
+    column: keyof DATA,
+    index: number
+  ) => string;
+}): JSX.Element {
+  const indexColumn = Object.keys(headers)[0];
+
+  return (
+    <div
+      className={`grid-table flex-1 grid-cols-[repeat(var(--cols),auto)] rounded print:p-1 ${className}`}
```

Comment:
(optional) this line is too long. split into multiple to be below <80 char limit